mne/morph.py

```diff
@@ -0,0 +1,953 @@
+# Author(s): Tommy Clausner <tommy.clausner@gmail.com>
+
+# License: BSD (3-clause)
+
+
+import os.path as op
+import warnings
+
+import numpy as np
+from scipy import sparse
+from scipy.sparse import block_diag as sparse_block_diag
+
+from . import read_forward_solution
+from .externals.h5io import read_hdf5, write_hdf5
+from .externals.six import string_types
+from .source_estimate import (VolSourceEstimate, SourceEstimate,
+                              VectorSourceEstimate)
+from .source_space import SourceSpaces
+from .surface import read_surface, read_morph_map, mesh_edges
+from .utils import (logger, verbose, get_subjects_dir, warn as warn_,
+                    has_nibabel, has_dipy)
+
+
+class SourceMorph(object):
+    """Container for source estimate morphs.
+
+    Parameters
+    ----------
+    src : instance of SourceSpaces
+        Information about the respective source space, with src being the
+        corresponding list of SourceSpaces
+    subject_from : str | None
+        Name of the original subject as named in the SUBJECTS_DIR.
+        If None src[0]['subject_his_id]' will be used. The default is None.
+    subject_to : str
+        Name of the subject on which to morph as named in the SUBJECTS_DIR
+        The default is 'fsaverage'.
+    data_from : array | list of two arrays | str | None
+        If morphing a surface source estimate, data_from is the
+        Vertex numbers corresponding to the data.
+        If morphing a volume source estimate data_from is the absolute path
+        where subject_from/mri/brain.mgz is stored
+        If None data_from will be found through subject_from if present. The
+        default is None.
+    data_to : array | list of two arrays | str | None
+        If morphing a surface source estimate , data_to is the
+        Vertex numbers corresponding to the data.
+        If morphing a volume source estimate data_to is the absolute path
+        where subject_to/mri/brain.mgz is stored
+        If None data_to will be found through subject_to if present. The
+        default is None.
```

Comment:
I don't get why you need data_from and data_to parameters? it's not something that can be obtained from src param?

Reply:
well it's because if you want to e.g. use a different volume then in ``'sample'``, you would have no chance to specify it... (Same for vertices). Using ``data_from`` and ``data_to`` you can force the morph to be performed on two data sets freely specifiable (vertices for surf and mri_paths for vol)

Reply:
I don't get what you mean sorry. Want to chat on gitter or hangout?

Reply:
texted you on handout